src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
 
-## Primitive Types
-
-Any of these types can be private (the default) or public (via a `pub` modifier).
-
-Note that private types are also referred to as witnesses.
+A **private value** is known only to the Prover, while a **public value** is known by both the Prover and Verifier. All public and private values are primitive types, though individual fields of compound types may also be marked private or public.
 
-#### Field Type
+> **Note:** For Noir programs using public smart contract verifiers, public values defined in such programs can be considered as public knowledge once their proofs are verified on-chain.
 
-A field type corresponds to a native field type in the backend. Usually this is a (roughly) 256-bit integer.
-This should generally be the default type reached for to solve problems. Using a smaller integer type like `u64` incurs
-extra range constraints and so is less efficient rather than more.
-
-```rust,noplaypen
-fn main(x : Field, y : Field) {
-    let z = x + y;
-}
-```
+Public data types are treated no differently to private types apart from the fact that their values will be revealed in proofs generated. Simply changing the value of a public type will not change the circuit (where the same goes for changing values of private types as well).
 
-`x`, `y` and `z` are witness types (private `Field`). Using the `let` keyword we derived a new witness `z` which is constrained to be equal to `x + y`.
+_Private_ are also referred to as _witnesses_ sometimes.
 
-#### Integer Type
+> **Note:** The terms private and public when applied to a type (e.g. `pub Field`) have a different meaning than when applied to a function (e.g. `pub fn foo() {}`).
+>
+> The latter is a visibility modifier familiar to those coming from other languages, while the former can be thought of as a visibility modifier for the Verifier to interpret.
 
-An integer type is a witness type which has been constrained using a range constraint.
-The Noir front-end currently supports arbitrary sized integer types.
+### pub Modifier
 
-Below we show the integer type in action:
+All data types in Noir are private by default. Types are explicitly declared as public using the `pub` modifier:
 
 ```rust,noplaypen
-fn main(x : Field) {
-    let y = x as u24;
+fn main(x : Field, y : pub Field) -> pub Field {
+    x + y
 }
 ```
 
-`x` and `y` are both private types (witnesses), however `y` is an integer type.
-If `y` exceeds the range \\([0,2^{24}-1]\\) then any proof created will output false by the verifier.
+In this example, the first input (i.e. `x`) is private while the second input (i.e. `y`) and the return value (i.e. `x + y`) are public.
 
-> **Note:** The Aztec backend supports even and odd sized integer types, so while using the Aztec backend, even sized integer types such as u32, u48, as well as odd sized integer types such as u5, u3 will produce proofs.
+> **Note:** As of this moment, public types can only be declared through parameters on `main`.
```

Comment:
Lets remove "As of this moment." All information in the docs is as of this moment and subject to change